### PR TITLE
Converted menu links into logos

### LINKED ISSUE #39: [Suggestion] Replace header link text by logos
**:information_source: If you agree with this design idea, please leave comment or add a :+1:**

---

### Kind of issue 

- [ ] Feature
- [ ] Bug
- [x] Design
- [ ] Other, namely:

### Description

> Related issues: https://github.com/simple-icons/simple-icons-website/issues/27

Currently, the website header contains three links to external platforms (and an about link), namely: GitHub, NPM, and Packagist. These links are in text, but it is common for such links to be icons - and so did many redesign mockups in https://github.com/simple-icons/simple-icons/issues/980. 

So the question is, should we replace these text links by icon links? I think I would be in favor of this as I think it looks better, it is arguably easier to parse an image than it is text. On the other hand, it is usually recommended to use text for accessibility - though workarounds for that exist.

Alternatively, we could also choose to have both text and icons, hiding the text in some kind of [compact mode](https://github.com/simple-icons/simple-icons-website/issues/17).

### LINKED ISSUE #58: Consider design for link to DISCLAIMER.md
### Kind of issue 

- [ ] Feature
- [ ] Bug
- [x] Design
- [ ] Other, namely: _requirement_

### Description

~~Add a link to the `DISCLAIMER.md` of the project - depends on https://github.com/simple-icons/simple-icons/pull/4912.~~

In #74 we were not necessarily happy about the design aspect of the link to the disclaimer, but we merged it in in the interest of having some link while we figure out the design. This ticket now serves to keep this topic alive and potentially discuss the design of the link to the disclaimer.

### PR DESCRIPTION
**Pull request by jorgeamadosoria**
- Converted menu links to icons:
image

image

Links included now are, in order from left to right:

- Github repository for simple-icons
- npm package
- packagist
- jsDelivr
- Unpkg
- Open Collective
- Legal Disclaimer
- About

Taken mostly from here: https://petershaggynoble.github.io/SI-Sandbox/

Unpkg icon taken (invented?) from https://twitter.com/unpkg
Apparently there is/was a rebranding effort, which is good since Unpkg does not seem to have a logo of their own.

The legal disclaimer icon is a simple scale made wider to allow for light/dark mode switch. However it is visually awful. A better icon would be preferred, specially in dark mode.

The menu is non responsive, as the text menu wasn't either.

@PeterShaggyNoble's sandbox menu is responsive. I think we should copy more things from there, including the npm package command. A responsive menu would be a nice touch.

### CODE REVIEW

File: public/index.pug
Comment (ericcornelissen) [inline public/index.pug]: Wondering if writing "Content Delivery Network" in full here makes more sense, space is not a concern anyway

```suggestion
          a(class="header__list-icon" title="jsDelivr (Content Delivery Network)" href="https://www.jsdelivr.com/package/npm/simple-icons" rel="noopener") 
```

File: public/index.pug
Comment (ericcornelissen) [inline public/index.pug]: Wondering if writing "Content Delivery Network" in full here makes more sense, space is not a concern anyway

```suggestion
          a(class="header__list-icon" title="Unpkg (Content Delivery Network)" href="https://unpkg.com/browse/simple-icons/" rel="noopener") 
```

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: The buttons are currently missing an effect when hovered, it should probably just be a lighter color (or darker color and make the standard color lighter). Relatedly, I would say the `:focus` styling should be set to the same as the `:hover` styling and have the outline removed.

Also, this color is currently the same as `--dm-color-button-hover` (same for the `--lm` variants) so we should probably re-use those in some way. Do you feel like you can do that? This stylesheet is still a bit of a mess in some regards (including this regard) so I can help you out here if you prefer.

File: tests/e2e.test.js
Comment (ericcornelissen) [inline tests/e2e.test.js]: Love the improvement 👍 Just one suggestion for clarity:

```suggestion
  const menuLinksTitles = [
    'repository',
    'npm',
    'packagist',
    'jsDelivr (CDN)',
    'Unpkg (CDN)',
    'Open Collective',
    'Legal disclaimer',
    'About Simple Icons',
  ];

  menuLinksTitles.forEach((title) =>
```

File: public/index.pug
Comment (ericcornelissen) [inline public/index.pug]: I think the `stroke` should be removed here, it [looks a little bit weird in dark mode](https://user-images.githubusercontent.com/15719479/130332670-c5f7b10d-713e-4380-b10d-8ee9ac7d97ed.png)

```suggestion
            svg(role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg")
```

File: public/index.pug
Comment (ericcornelissen) [inline public/index.pug]: Looks like this SVG could be optimized with a lower precisions 🤔

File: public/index.pug
Comment (ericcornelissen) [inline public/index.pug]: We _could_ remove it from the footer now, what do you think?

**Review by ericcornelissen — CHANGES_REQUESTED** [review]
Looking good already, nice work. I have some requests for changes but they're mostly minor. I also added to the title that it adds "Open Collective".

The biggest thing I noticed is that it currently doesn't work in Firefox (the icons are not scaled down there), can you look into this?


> The legal disclaimer icon is a simple scale made wider to allow for light/dark mode switch. However it is visually awful. A better icon would be preferred, specially in dark mode.

I agree that a different icon would be preferred (though it being a scale is fine). In the long run it should probably come from the same set as will be used to resolve #50. That said,  I'd be fine merging this PR as is and opening an issue to update the icon (or add it as part of #50).


> I think we should copy more things from there, including the npm package command. A responsive menu would be a nice touch.

I'd recommend you to open (two separate) issues for that - even if you currently plan to resolve them in this Pull Request.

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: I guess I can add them. Agreed on the reuse of button hover and focus

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug:52]: agreed. will do

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: true. done.

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: Initially I thought you meant the disclaimer.
I would keep the Open Collective link in both footer and header. It's not going to hurt to have "calls to action" in a couple places. And it's not obtrusive or anything in the footer either, so... I say let it. At least until we need the space for something else more important.

But since you mentioned the footer.... should we add twitter to the menu as well?

Comment (ericcornelissen) [reply public/index.pug]: Just a suggestion (hence the italics _could_), I'm fine either way.

> But since you mentioned the footer.... should we add twitter to the menu as well?

I'm a bit on the fence in that regard: all the other links are to _our_ page on the respective platform (excluding the disclaimer), whereas the twitter button in the footer is to tweet about the project which can at least be confusing and may be interpreted as being misleading (I personally wouldn't like it). What do you think?

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: uff, yes. what an oversight on my part.

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: I agree. It would look out of place in the menu.
Let's keep it down there for the time being

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: added an effect on hover. I also checked it on firefox, it looks ok to me.
The pixel values are totally arbitrary, open to changes if you would like more / less margin or bigger / smaller menu icons

**Comment by jorgeamadosoria** [comment]
Done with the changes. when we have this one merged I will add the report icon button, and after that, I'll create the ticket for responsive menu. 
In that order.

File: public/index.pug
Comment (ericcornelissen) [inline public/index.pug]: What's the motivation for adding this link? I'm not sure what purpose it serves in addition to the GitHub link

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: Interesting choice for a hover/focus effect. I would personally prefer something more similar to existing hover effects (none of which scale the hover target). E.g. change the color of the icon or invert it and add a background. What do you think? If you'd like to stick with what you have now, I think we should ask for input from other maintainers.

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: Am I mistaken or is this rule redundant given [line 284](https://github.com/simple-icons/simple-icons-website/blob/fdf6e6e4a60522671c27832831395cb142de9d70/public/stylesheet.css#L284) :thinking:

**Comment by ericcornelissen** [comment]
Also, is this still a WIP/draft @jorgeamadosoria - if not could you update the title and mark it as ready for review.
Stil a WIP because of the hover effect, which links were going to be added or left in the footer, etc. 
Let's agree on all that first, then we can have it ready for review by anyone else.

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: It's different to point to the repository than to point to the README file.
Besides, the idea is to point to the contributing page, the contributors list, maybe put the legal disclaimer in there as well.

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: I started with the inverse color effect. the problem is that it looks good in the buttons because they have borders. Without borders in the menu logos (which looks ugly) the other choice is to have a round "halo" to serve as background when inverting the color. It's never exactly the same as the buttons hover effect, so this one was faster.
I can go back to inverting the colors, if we are ok living with the circle color to serve as background.
Or any other idea to provide a background for the inverted icon.

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: that depends on whether we are going to go back to inverted colors.

File: public/index.pug
Comment (ericcornelissen) [reply public/index.pug]: I didn't say it's the same. I'm asking what the point of it is in addition to the existing link. If you want it to point to the readme than we can change the existing link.

If the ultimate goal is to do something else, than I don't see a reason to add this as an intermediate step.

File: public/stylesheet.css
Comment (ericcornelissen) [reply public/stylesheet.css]: I'm not saying it has to be the same effect as the control buttons, after all there are more interactive elements on the page.

I think the simplest (and best) solution would be to change the color (either make the default gray and hover black or the other way around). But I'm definitely open to suggestions.

File: public/stylesheet.css
Comment (ericcornelissen) [reply public/stylesheet.css]: So, it is redundant at the moment? It's hard to comment based on a potential change in the future... It's easier to review of the Pull Request is in a consistent state. (Admittedly the PR is still in a draft state but your last comment suggester to me it's ready for review)

Comment (jorgeamadosoria) [reply public/stylesheet.css]: yes, at the moment it is redundant and can be removed. It is in there because I suspected the color inversion was going to be asked to put back in.
That's also why this is a WIP: it's not ready for merging until we iron out all of these details. I wouldn't want accidents to happen again.

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: well, the most visually pretty effect is inverting the colors with the background. But the problem is whether we can live with the circular background.
Or rounded square background, same problem.
Grey could work, but it reduces the contrast with the background color in both light and dark cases, That doesn't help the UX. Better to have high contrast by default (white on black, black on white). Making the hover grey could work, I would have to see how it looks. I can try it.
If you want, I can push a circular background version so you can see it. Maybe I see it being uglier than you, and I'm being too harsh on that variant, idk.

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: the ultimate point of this link, right now, is to point to the readme file rather than the root of the repository.
At a later time, this can be evolved to be a dropdown with more links. Or not. If it stays like this, I'm good.
It's not an intermediate step that has to evolve to something else, it's an intermediate step that can become a final step if we choose to. The menu is stable as it is.
Regardless, I'm not married to it. If you want to remove it, we can. There are more important things to do.

Comment (ericcornelissen) [reply public/index.pug]: I'm just asking what value it adds right now. I'm not saying this should be removed before this PR can be merged. And again I can tell the difference, no need to point that out. If you think it makes sense to have it by all means leave it in. I would like to point your attention to #27 as well.

As for "intermediate" aspect, I personally don't see a reason to add something now because we might want to add something later, while that something we want to add later doesn't _require_ the thing we're adding now.

File: public/stylesheet.css
Comment (ericcornelissen) [reply public/stylesheet.css]: > the most visually pretty effect is inverting the colors with the background

That is a matter of opinion. (And yes, I also said "the best solution is to change the color", but I prefaced that with "I think".)


> Grey could work, but it reduces the contrast with the background color in both light and dark cases, That doesn't help the UX. Better to have high contrast by default (white on black, black on white). Making the hover grey could work, I would have to see how it looks. I can try it.

As long as the contract is sufficient, I don't see a problem. Don't get me wrong, it's easier to get wrong, but it doesn't have to be.


> If you want, I can push a circular background version so you can see it. Maybe I see it being uglier than you, and I'm being too harsh on that variant, idk.

Only one way to find out, and that's to see what it actually looks like.

File: public/stylesheet.css
Comment (ericcornelissen) [reply public/stylesheet.css]: I understand now why you leave it as a draft, that makes sense.

However, that does not change the fact that I can only review the changes in the Pull Request as a whole. Again, you cannot expect the reviewer to know that you leave this in because maybe you need it later.

Also, if whether or not this comment is resolved depends on a future change, I don't think it should be marked as resolved yet.

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: The menu now has the background color effect as discussed.

File: public/index.pug
Comment (ericcornelissen) [inline public/index.pug]: This should link to the disclaimer, not the readme

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: This focus doesn't work because the focusable element is the `<a>` (i.e. `.header__list-icon`)

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: This seems redundant, tested it when removed and doesn't seem to affect anything :thinking:

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: These, and all related variables, are currently unused.

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: I think we can and should just reuse the `--color-button` variables for the "menu" item. If you want to have a separate variable - e.g. because we may want a different color in the header at some point - then this variable should for now reference the existing one to ensure they're the same as long as that's the goal.

Also, "menu" is a bit of an odd choice of word as it doesn't appear in the HTML.

**Review by ericcornelissen — CHANGES_REQUESTED** [review]

File: public/index.pug
Comment (ericcornelissen) [reply public/index.pug]: Okay, but that's unrelated to this comment thread.

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: it is necessary. Otherwise the fill does not go to the svg.
Did you try Chrome? That's what I use.

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: removed color-menu-icon-hover

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: removed color-menu-icon, reused color-button-active

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: I tried firefox, and it works without the fill there. Chrome requires it, however.

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: Link to README has been removed

**Comment by jorgeamadosoria** [comment]
Still something missing here, @ericcornelissen ? I'm only waiting on your review of the changes requested.

File: public/index.pug
Comment (ericcornelissen) [inline public/index.pug]: Could you remove the `rel="noopener"` here and in the other header links. If there's no `target` specified than it's not needed, see also #83.

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: This assumes the structure of the document a bit too much I would say, why not just use the `.header__list-icon` selector?

If you think that's not a good solution because styling a class named "icon" in this way is odd, then I'd suggest we introduce a new class for clarity (e.g. `.header__list-link`).

**Review by ericcornelissen — CHANGES_REQUESTED** [review]
The hovering effect seems off to me now without the padding. Why did you change that?

> I'm only waiting on your review of the changes requested.

Sorry, wasn't aware it could be reviewed again as it's still a draft.

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: removed

**Comment by jorgeamadosoria** [comment]
> The hovering effect seems off to me now without the padding. Why did you change that?
> 
> > I'm only waiting on your review of the changes requested.
> 
> Sorry, wasn't aware it could be reviewed again as it's still a draft.

readded padding.

File: public/index.pug
Comment (ericcornelissen) [reply public/index.pug]: Looks like you removed it everywhere, not only in these header links :confused: 

Also, minor (and unfortunate that Prettier doesn't pick this up), but now there's a space between the last quote and the closing parenthesis of the `.header__list-icon` elements.

Comment (jorgeamadosoria) [reply public/index.pug]: done correcting that.

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: I feel like it would make more sense to use the same selector twice, what do you think @jorgeamadosoria?

```suggestion
.header__list-icon:hover,
.header__list-icon:focus {
```

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: Why don't you just set the `fill` value directly here:

```suggestion
  fill: var(--color-button-active);
```

Rather then inheriting it from it's parent [where you specify the color without any effect(?) on line 271](https://github.com/simple-icons/simple-icons-website/blob/50f437a69847dd976ba02619b01f6902680b0cec/public/stylesheet.css#L271).

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: This used to be `10%` (e.g. in 2fd961def848b63c93c230566ebec660e029d027), why did you change it to `50%`? This surprised me.

First, I don't think the `50%` looks good with the npm link in particular:

image


More generally speaking, I don't feel like the circle suites the style of the rest of the website. Other elements have slightly rounded corners (e.g. search input, control buttons, cards) [specified in exact pixels](https://github.com/simple-icons/simple-icons-website/blob/cdc99f1be3f68911cbe69b5277daec936e136578/public/stylesheet.css#L24-L25). I would suggest we do the same here for consistency, e.g.:

```suggestion
  border-radius: var(--border-radius-large);
```

What do you think @jorgeamadosoria?

File: public/stylesheet.css
Comment (ericcornelissen) [inline public/stylesheet.css]: Is there any particular reason you repeated the exact pixel values here? Why not just do

```suggestion
  height: 100%;
  width: 100%;
```

**Review by ericcornelissen — CHANGES_REQUESTED** [review]

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css:275]: it's not possible: header__list-item is a <li> element, it has no focus

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: I don't know, less letters? it's the same effect. 
If you prefer it set directly, we can do it.

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: I don't have strong opinions one way or the other. I put 50% to make the background round. If you prefer it at 6px, that's fine too.

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: no particular reason. I was tweaking the values and it stayed like that.

File: public/stylesheet.css
Comment (ericcornelissen) [reply public/stylesheet.css]: > I don't know, less letters?

By my count having:

```css
[...]
fill: var(--color-button-active);
[...]
fill: inherit;
[...]
```

Is more letter than 

```css
fill: var(--color-button-active);
```

File: public/stylesheet.css
Comment (ericcornelissen) [reply public/stylesheet.css]: Just suggested it because now you can't mess up by changing the pixel value only in one place.

File: public/stylesheet.css
Comment (ericcornelissen) [reply public/stylesheet.css]: If you think `50%` is the way to go, please stick with that and we ship it that way :slightly_smiling_face:

Comment (jorgeamadosoria) [reply public/stylesheet.css]: no, no, it's fine. Like I said, I don't have a horse in this race. Let's go with yours.

Comment (jorgeamadosoria) [reply public/stylesheet.css]: I already committed the change, even.

File: public/stylesheet.css
Comment (jorgeamadosoria) [reply public/stylesheet.css]: it's fine, I applied the suggestion. It's better not to have duplicates.

**Comment by jorgeamadosoria** [comment]
I think that's all the changes?

File: public/index.pug
Comment (ericcornelissen) [inline public/index.pug]: Shouldn't this link to the `DISCLAIMER.md` file?

**Review by ericcornelissen — CHANGES_REQUESTED** [review]

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: huh. yes, yes it should. I didn't remove the correct link when removing the duplicate Readme, apparently.

Comment (ericcornelissen) [reply public/index.pug]: Feel free to merge once you fixed it.

**Review by ericcornelissen — APPROVED** [review]

**Pull request merged**
